typing/typeclass.ml

```diff
@@ -1190,7 +1190,7 @@ and class_expr_aux cl_num val_env met_env virt self_scope scl =
       in
       let partial =
         let dummy = type_exp val_env (Ast_helper.Exp.unreachable ()) in
-        Typecore.check_partial val_env pat.pat_type pat.pat_loc
+        Typecore.check_partial Modules_rejected val_env pat.pat_type pat.pat_loc
```

Comment:
The earlier call to `Typecore.type_class_arg_pattern` is what first typechecks the pattern (and this function does set `Modules_rejected`). This `check_partial` call is just for checking pattern exhaustiveness (I think?). It would probably be fine for `check_partial` to just always allow module patterns, but then you have to choose a scope to enter them at, which feels weird.